src/shapes/image.class.ts

```diff
@@ -1,869 +1,797 @@
-       * @type String
-       * @default
-       */
-      type: 'image',
-
-      /**
-       * Width of a stroke.
-       * For image quality a stroke multiple of 2 gives better results.
-       * @type Number
-       * @default
-       */
-      strokeWidth: 0,
-
-      /**
-       * When calling {@link fabric.Image.getSrc}, return value from element src with `element.getAttribute('src')`.
-       * This allows for relative urls as image src.
-       * @since 2.7.0
-       * @type Boolean
-       * @default
-       */
-      srcFromAttribute: false,
-
-      /**
-       * private
-       * contains last value of scaleX to detect
-       * if the Image got resized after the last Render
-       * @type Number
-       */
-      _lastScaleX: 1,
-
-      /**
-       * private
-       * contains last value of scaleY to detect
-       * if the Image got resized after the last Render
-       * @type Number
-       */
-      _lastScaleY: 1,
-
-      /**
-       * private
-       * contains last value of scaling applied by the apply filter chain
-       * @type Number
-       */
-      _filterScalingX: 1,
-
-      /**
-       * private
-       * contains last value of scaling applied by the apply filter chain
-       * @type Number
-       */
-      _filterScalingY: 1,
-
-      /**
-       * minimum scale factor under which any resizeFilter is triggered to resize the image
-       * 0 will disable the automatic resize. 1 will trigger automatically always.
-       * number bigger than 1 are not implemented yet.
-       * @type Number
-       */
-      minimumScaleTrigger: 0.5,
-
-      /**
-       * List of properties to consider when checking if
-       * state of an object is changed ({@link fabric.Object#hasStateChanged})
-       * as well as for history (undo/redo) purposes
-       * @type Array
-       */
-      stateProperties: FabricObject.prototype.stateProperties.concat(
-        'cropX',
-        'cropY'
-      ),
+import { fabric } from '../../HEADER';
+import type { BaseFilter } from '../filters/base_filter.class';
+import { getFilterBackend } from '../filters/FilterBackend';
+import { SHARED_ATTRIBUTES } from '../parser/attributes';
+import { parseAttributes } from '../parser/parseAttributes';
+import { TClassProperties, TSize } from '../typedefs';
+import { cleanUpJsdomNode } from '../util/dom_misc';
+import { createCanvasElement } from '../util/misc/dom';
+import { findScaleToCover, findScaleToFit } from '../util/misc/findScaleTo';
+import {
+  enlivenObjectEnlivables,
+  enlivenObjects,
+  loadImage,
+  LoadImageOptions,
+} from '../util/misc/objectEnlive';
+import { parsePreserveAspectRatioAttribute } from '../util/misc/svgParsing';
+import { FabricObject, fabricObjectDefaultValues } from './fabricObject.class';
+
+export type ImageSource =
+  | HTMLImageElement
+  | HTMLVideoElement
+  | HTMLCanvasElement;
+
+/**
+ * @tutorial {@link http://fabricjs.com/fabric-intro-part-1#images}
+ */
+export class Image extends FabricObject {
```

Comment:
this name is a pain as it is Object now. doing new Image() can't be really done.

Reply:
I agree, wasn't sure what to do cause if we name it FabricImage then maybe every thing should become FabricXXX